client-addon/src/bootstrap.js

```diff
@@ -54,13 +62,14 @@ function stopObserving() {
 
 var siteSet = new Set();
 var warnedHostSet = new Set();
+var breachData = {}
```

Comment:
Instead of having a new object, we can change siteSet to be a Map (breachData is a good name; let's keep it). Then, whenever we need to get the data for a host, we can do (for example)

```
let data = breachData.get(host);
if (!warnedHostSet.has(host) || !data || !data.verified) {
  return;
}
// Etc
```

Apart from that, here's a nit: please terminate statements with a semicolon. :)

Reply:
> Apart from that, here's a nit: please terminate statements with a semicolon. :)

Hah 😉 I probably should have noticed you were doing that in surrounding code – I never know what people prefer in languages where it is optional ;p